Fatal org.elasticsearch.bootstrap.JNAKernel32Library error starting Elasticsearch 5.3.0 snapshot on Windows 10

I am getting a fatal error when attempting to start elasticsearch on Windows 10. See console output below.

This is with the snapshot dated Sun Feb 05 21:59:58 GMT

C:\Tools\elasticsearch-5.3.0-SNAPSHOT\bin>elasticsearch.bat
[2017-02-06T10:41:39,870][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [] fatal error in thread [main], exiting
java.lang.ExceptionInInitializerError: null
        at org.elasticsearch.bootstrap.JNAKernel32Library.getInstance(JNAKernel32Library.java:68) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.JNANatives.addConsoleCtrlHandler(JNANatives.java:201) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Natives.addConsoleCtrlHandler(Natives.java:84) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.initializeNatives(Bootstrap.java:125) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:204) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:360) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:114) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:58) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:122) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.cli.Command.main(Command.java:88) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
Caused by: java.lang.IllegalArgumentException: Not allowed to create an instance of class org.elasticsearch.bootstrap.JNAKernel32Library$SizeT, requires a public, no-arg constructor: java.lang.IllegalAccessException: Class com.sun.jna.NativeMappedConverter can not access a member of class org.elasticsearch.bootstrap.JNAKernel32Library$SizeT with modifiers ""
        at com.sun.jna.NativeMappedConverter.defaultValue(NativeMappedConverter.java:60) ~[jna-4.2.2.jar:4.2.2 (b0)]
        at com.sun.jna.NativeMappedConverter.<init>(NativeMappedConverter.java:44) ~[jna-4.2.2.jar:4.2.2 (b0)]
        at com.sun.jna.NativeMappedConverter.getInstance(NativeMappedConverter.java:32) ~[jna-4.2.2.jar:4.2.2 (b0)]
        at com.sun.jna.Native.register(Native.java:1610) ~[jna-4.2.2.jar:4.2.2 (b0)]
        at com.sun.jna.Native.register(Native.java:1529) ~[jna-4.2.2.jar:4.2.2 (b0)]
        at com.sun.jna.Native.register(Native.java:1252) ~[jna-4.2.2.jar:4.2.2 (b0)]
        at org.elasticsearch.bootstrap.JNAKernel32Library.<init>(JNAKernel32Library.java:57) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.JNAKernel32Library.<init>(JNAKernel32Library.java:41) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        at org.elasticsearch.bootstrap.JNAKernel32Library$Holder.<clinit>(JNAKernel32Library.java:51) ~[elasticsearch-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        ... 13 more

--- a/buildSrc/src/main/resources/checkstyle_suppressions.xml
+++ b/buildSrc/src/main/resources/checkstyle_suppressions.xml
@@ -200,6 +200,7 @@
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]action[/\\]update[/\\]UpdateRequestBuilder.java" checks="LineLength" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]bootstrap[/\\]Bootstrap.java" checks="LineLength" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]bootstrap[/\\]JNANatives.java" checks="LineLength" />
+  <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]bootstrap[/\\]JNAKernel32Library.java" checks="RedundantModifier" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]bootstrap[/\\]JVMCheck.java" checks="LineLength" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]bootstrap[/\\]JarHell.java" checks="LineLength" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]bootstrap[/\\]Security.java" checks="LineLength" />

--- a/core/src/main/java/org/elasticsearch/bootstrap/JNAKernel32Library.java
+++ b/core/src/main/java/org/elasticsearch/bootstrap/JNAKernel32Library.java
@@ -155,7 +155,7 @@ final class JNAKernel32Library {
 
     public static class SizeT extends IntegerType {
 
-        SizeT() {
+        public SizeT() {
             this(0);
         }
 

--- a/core/src/main/java/org/elasticsearch/bootstrap/JNAKernel32Library.java
+++ b/core/src/main/java/org/elasticsearch/bootstrap/JNAKernel32Library.java
@@ -155,6 +155,8 @@ final class JNAKernel32Library {
 
     public static class SizeT extends IntegerType {
 
+        // JNA requires this no-arg constructor to be public,
+        // otherwise it fails to register kernel32 library
         public SizeT() {
             this(0);
         }

